### PR TITLE
LS-455-gke-nav-and-doc

### PR DESCRIPTION
**Pull request by mendonk**

### CODE REVIEW

File: modules/ROOT/pages/install-gke.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+See this https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[values file] for more details on the settings in `values.yaml`.
+
+Additional provisioning set-up options are available https://docs.k8ssandra.io/install/gke/[here].
+
+== Install Luna Streaming to GKE
+
+Once you have added your storage settings to the `storage-values.yaml` file, install the Helm chart. 
+
+. Create the namespace; in this example, we use `pulsar`.
```
Comment (yabinmeng) [inline modules/ROOT/pages/install-gke.adoc:40]: Don't think this is needed

File: modules/ROOT/pages/install-gke.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+. To avoid having to specify the `pulsar` namespace on each subsequent command, set the namespace context. Example:
++
+----
+kubectl config set-context $(kubectl config current-context) --namespace=pulsar
+----
+
+. Execute `kubectl get pods -A` to view your running pods. 
+. Once your pods are up and running, continue to xref:quickstart-helm-installs.adoc#manage-pulsar-cluster[Accessing Pulsar Cluster in Cloud] to access and manage your cluster. 
+
+== Next
```
Comment (yabinmeng) [inline modules/ROOT/pages/install-gke.adoc:61]: As a general comment, especially after I also took a look at the other 2 cloud provider specific documents, I don' think these documents are really necessary beyond the existing Helm Chart doc: https://docs.datastax.com/en/luna/streaming/2.8/quickstart-helm-installs.html

The only difference in these documents is about the storage class difference which is already covered by the Helm Chart doc. Other than that, there is nothing cloud provider specific stuff. 

So unless we want to describe the procedure differences about how to launch the K8s clusters in these different cloud providers, I don't think there are much value added by having these 3 documents.

File: modules/ROOT/pages/install-gke.adoc [hunk 3]
```diff
@@ -0,0 +1,109 @@
+= Luna Streaming Installation on Google Kubernetes Environment
+
+This document will cover installation of Luna Streaming on the Google Kubernetes Environment, or GKE.
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:3]:
```suggestion
This document will cover installation of Luna Streaming on the Google Kubernetes Environment (GKE). 
```

File: modules/ROOT/pages/install-gke.adoc [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+For an example set of production cluster values, see the DataStax production-ready https://github.com/datastax/pulsar-helm-chart[Helm chart]. +
+
+DataStax recommends these hardware resources for running Luna Streaming in a Kubernetes environment: +
+
+include::partial$helm-chart-recommendations.adoc[]
+
+For the local machine running the Helm chart, you will need: +
+
+* https://helm.sh/docs/[Helm] version 3 +
+* A Google account with priveleges to install and configure Kubernetes resources.
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:14]:
```suggestion
* A Google account with privileges to install and configure Kubernetes resources. 
```

File: modules/ROOT/pages/install-gke.adoc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+include::partial$helm-chart-recommendations.adoc[]
+
+For the local machine running the Helm chart, you will need: +
+
+* https://helm.sh/docs/[Helm] version 3 +
+* A Google account with priveleges to install and configure Kubernetes resources. 
+
+== Helm chart `values.yaml`
+
+The https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[values.yaml] will deploy Pulsar with:
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:18]:
```suggestion
The https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[`values.yaml`] will deploy Pulsar with:
```

File: modules/ROOT/pages/install-gke.adoc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+* 1 ZooKeeper cluster with `replicaCount:3` +
+* 1 Bookkeeper cluster with `replicaCount:3` +
+* 1 Broker cluster with `replicaCount: 3` +
+** Ledgers: `defaultEnsembleSize: 2`, `defaultAckQuorum: 2`, `defaultWriteQuorum: 2` +
+* 1 Function cluster with a `replicaCount: 2` +
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:24]:
```suggestion
* 1 ZooKeeper cluster with `replicaCount:3`
* 1 Bookkeeper cluster with `replicaCount:3`
* 1 Broker cluster with `replicaCount: 3`
** Ledgers: `defaultEnsembleSize: 2`, `defaultAckQuorum: 2`, `defaultWriteQuorum: 2`
* 1 Function cluster with a `replicaCount: 2`
```

File: modules/ROOT/pages/install-gke.adoc [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+=== `default_storage`
+The `default_storage` parameter in `values.yaml` controls the default storage class for all persistent volumes created by the Helm chart. +
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:27]:
```suggestion
The `default_storage` parameter in `values.yaml` controls the default storage class for all persistent volumes created by the Helm chart.
```

File: modules/ROOT/pages/install-gke.adoc [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+[source,yaml]
+----
+default_storage:
+  existingStorageClassName: default
+----
+
+For a component like BookKeeper, which requires *stateful* storage, we need to override the `default_storage` class when the BookKeeper Persistent Volume Claims (PVCs) are created. +
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:35]:
```suggestion
For a component like BookKeeper, which requires stateful storage, we need to override the `default_storage` class when the BookKeeper Persistent Volume Claims (PVCs) are created.
```

File: modules/ROOT/pages/install-gke.adoc [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+There are two ways to override `default_storage`: +
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:37]:
```suggestion
There are two ways to override `default_storage`:
```

File: modules/ROOT/pages/install-gke.adoc [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+1. Leave `existingStorageClassName` blank and specify the storage class parameters below. +
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:39]:
```suggestion
* Leave `existingStorageClassName` blank and specify the storage class parameters below.
```

File: modules/ROOT/pages/install-gke.adoc [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        existingStorageClassName: 
+        storageClass:
+         provisioner: kubernetes.io/gce-pd
+         type: pd-ssd
+         fsType: ext4
+         extraParams:
+            replication-type: none
+----
+
+2. Create a custom storage configuration as a `yaml` file (https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/templates/bookkeeper/bookkeeper-storageclass.yaml[like this one]) and tell the Helm chart to use that storage configuration when it creates the BookKeeper PVCs. +
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:56]:
```suggestion
* Create a custom storage configuration as a `yaml` file (https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/templates/bookkeeper/bookkeeper-storageclass.yaml[like the DataStax example]) and tell the Helm chart to use that storage configuration when it creates the BookKeeper PVCs.
```

File: modules/ROOT/pages/install-gke.adoc [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+#   extraParams:
+#      replication-type: none
+----
+
+See this https://github.com/datastax/pulsar-helm-chart/blob/master/helm-chart-sources/pulsar/values.yaml[values file] for more details on the settings in `values.yaml`.
+
+Additional provisioning set-up options are available https://docs.k8ssandra.io/install/gke/[here].
+
+== Install Luna Streaming to GKE
+
```
[the remark this answers is not in the document]
Comment (jgillenwater) [reply modules/ROOT/pages/install-gke.adoc:82]: @mendonk Please make these changes or resolve if you already have. Thanks!

File: modules/ROOT/pages/install-gke.adoc (hunk 1 above)
Comment (jgillenwater) [reply modules/ROOT/pages/install-gke.adoc:85]: @mendonk Please make these changes or resolve if you already have. Thanks!

File: modules/ROOT/pages/install-gke.adoc (hunk 2 above)
Comment (jgillenwater) [reply modules/ROOT/pages/install-gke.adoc:106]: @mendonk Please make these changes or resolve if you already have. Thanks!

File: modules/ROOT/pages/install-gke.adoc [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  volumes:
+    journal:
+      name: journal
+      size: 20Gi
+        existingStorageClassName: bookkeeper-storageclass.yaml
+----
+
+=== GKE-specific storage class 
+
+Here is an example `storage-values.yaml` as used above for the BookKeeper node. Paste it into a `yaml` file to provision a persistent SSD with the ext4 file system.
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:69]:
```suggestion
Here is an example `storage-values.yaml` as used above for the BookKeeper node. Paste it into a `yaml` file to provision a persistent SSD with the `ext4` file system.
```

File: modules/ROOT/pages/install-gke.adoc [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+. To avoid having to specify the `pulsar` namespace on each subsequent command, set the namespace context. Example:
++
+----
+kubectl config set-context $(kubectl config current-context) --namespace=pulsar
+----
+
+. Execute `kubectl get pods -A` to view your running pods. 
+. Once your pods are up and running, continue to xref:quickstart-helm-installs.adoc#manage-pulsar-cluster[Accessing Pulsar Cluster in Cloud] to access and manage your cluster. 
+
+== Next
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:106]:
```suggestion

== What's next?
```

File: modules/ROOT/pages/install-gke.adoc [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+
+To learn more about using the Pulsar Admin Console, see xref:admin-console-tutorial.adoc[Admin Console Tutorial].
+To learn more about installing Pulsar on a server or virtual machine, see xref:quickstart-server-installs.adoc[Quickstart for Server/VM Install].
```
Comment (jgillenwater) [inline modules/ROOT/pages/install-gke.adoc:109]:
```suggestion
Once your pods are up and running, continue to xref:quickstart-helm-installs.adoc#manage-pulsar-cluster[Accessing Pulsar Cluster in Cloud] to access and manage your cluster. 

* To learn more about using the Pulsar Admin Console, see xref:admin-console-tutorial.adoc[Admin Console Tutorial].
* To learn more about installing Pulsar on a server or virtual machine, see xref:quickstart-server-installs.adoc[Quickstart for Server/VM Install].
```

**Review by jgillenwater — CHANGES_REQUESTED** [review]

**Pull request merged**